xbmc/network/Socket.h

```diff
@@ -70,21 +75,56 @@ namespace SOCKETS
 
     void SetAddress(const char *address)
     {
+      in6_addr addr6;
       memset(&saddr, 0, sizeof(saddr));
-      saddr.sin_family = AF_INET;
-      saddr.sin_addr.s_addr = inet_addr(address);
-      size = sizeof(saddr);
+      if (inet_pton(PF_INET6, address, &addr6) == 1)
```

Comment:
As on other comment I chose the simplest path to limit changes to ensure quick validation.